[libui] Undefined symbols on macOS 11.1 (20C69)

**Is your feature request related to a problem? Please describe.**

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES C)

set(src src/main.c)

find_package(unofficial-libui CONFIG REQUIRED)

add_executable(${PROJECT_NAME} ${src})
target_link_libraries(${PROJECT_NAME} PRIVATE unofficial::libui::libui)
````

### `src/main.c`
https://raw.githubusercontent.com/andlabs/libui/d788d86/examples/drawtext/main.c but with `#include "../../ui.h"` replaced with `#include <ui.h>`

### `shell`
```sh
$ [path]/vcpkg install libui
$ mkdir build && cd $_
$ cmake .. -DCMAKE_TOOLCHAIN_FILE=[path]/vcpkg/scripts/buildsystems/vcpkg.cmake
$ cmake --build .
[ 50%] Linking C executable pp
Undefined symbols for architecture x86_64:
  "_CFArrayAppendValue", referenced from:
      _addCTFeatureEntry in libui.a(opentype.m.o)
  "_CFArrayCreateMutable", referenced from:
      _uiprivOpenTypeFeaturesToCTFeatures in libui.a(opentype.m.o)
  "_CFArrayGetCount", referenced from:
      _matchStyle in libui.a(fontmatch.m.o)
      _uiprivMakeVariationAxisDict in libui.a(fontvariation.m.o)
  "_CFArrayGetValueAtIndex", referenced from:
      _matchStyle in libui.a(fontmatch.m.o)
      _uiprivMakeVariationAxisDict in libui.a(fontvariation.m.o)
  "_CFAttributedStringBeginEditing", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
  "_CFAttributedStringCreate", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
  "_CFAttributedStringCreateMutableCopy", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
  "_CFAttributedStringEndEditing", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
  "_CFAttributedStringGetAttribute", referenced from:
      _applyFontAttributes in libui.a(attrstr.m.o)
      _addFontAttributeToRange in libui.a(attrstr.m.o)
  "_CFAttributedStringGetLength", referenced from:
      -[uiprivTextFrame initWithLayoutParams:] in libui.a(drawtext.m.o)
      _applyFontAttributes in libui.a(attrstr.m.o)
  "_CFAttributedStringRemoveAttribute", referenced from:
      _applyFontAttributes in libui.a(attrstr.m.o)
  "_CFAttributedStringSetAttribute", referenced from:
      _processAttribute in libui.a(attrstr.m.o)
      _applyFontAttributes in libui.a(attrstr.m.o)
      _addFontAttributeToRange in libui.a(attrstr.m.o)
      _addBackgroundAttribute in libui.a(attrstr.m.o)
  "_CFDataGetBytePtr", referenced from:
      _avarExtract in libui.a(fontvariation.m.o)
      _trySecondaryOS2Values in libui.a(fonttraits.m.o)
  "_CFDataGetLength", referenced from:
      _trySecondaryOS2Values in libui.a(fonttraits.m.o)
  "_CFDictionaryAddValue", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
      _uiprivFontDescriptorToCTFontDescriptor in libui.a(fontmatch.m.o)
  "_CFDictionaryCreate", referenced from:
      _uiprivCTFontDescriptorAppendFeatures in libui.a(fontmatch.m.o)
      _addCTFeatureEntry in libui.a(opentype.m.o)
  "_CFDictionaryCreateMutable", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
      _uiprivFontDescriptorToCTFontDescriptor in libui.a(fontmatch.m.o)
  "_CFDictionaryGetValue", referenced from:
      -[uiprivFontStyleData prepare] in libui.a(fontmatch.m.o)
      _extractAxisDictValue in libui.a(fontvariation.m.o)
      _uiprivMakeVariationAxisDict in libui.a(fontvariation.m.o)
      _tryAxis in libui.a(fontvariation.m.o)
  "_CFEqual", referenced from:
      _trySecondaryOS2Values in libui.a(fonttraits.m.o)
  "_CFNumberCreate", referenced from:
      _processAttribute in libui.a(attrstr.m.o)
      _uiprivFontDescriptorToCTFontDescriptor in libui.a(fontmatch.m.o)
      _addCTFeatureEntry in libui.a(opentype.m.o)
  "_CFNumberGetValue", referenced from:
      -[uiprivFontStyleData prepare] in libui.a(fontmatch.m.o)
      _extractAxisDictValue in libui.a(fontvariation.m.o)
      _tryAxis in libui.a(fontvariation.m.o)
  "_CFRelease", referenced from:
      -[uiprivTextFrame dealloc] in libui.a(drawtext.m.o)
      -[uiprivFontButton getfontdesc:] in libui.a(fontbutton.m.o)
      -[uiprivCombinedFontAttr toCTFontWithDefaultFont:] in libui.a(attrstr.m.o)
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
      _processAttribute in libui.a(attrstr.m.o)
      _applyFontAttributes in libui.a(attrstr.m.o)
      _mkcolor in libui.a(attrstr.m.o)
      ...
  "_CFRetain", referenced from:
      -[uiprivFontStyleData initWithFont:] in libui.a(fontmatch.m.o)
      -[uiprivFontStyleData initWithDescriptor:] in libui.a(fontmatch.m.o)
      _matchStyle in libui.a(fontmatch.m.o)
  "_CFStringCreateWithCString", referenced from:
      _uiprivFontDescriptorToCTFontDescriptor in libui.a(fontmatch.m.o)
      _otfArrayForEachOT in libui.a(opentype.m.o)
  "_CFStringCreateWithCharacters", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
  "_CFStringCreateWithCharactersNoCopy", referenced from:
      _uiprivNewGraphemes in libui.a(graphemes.m.o)
  "_CFStringFind", referenced from:
      _guessItalicOblique in libui.a(fontmatch.m.o)
  "_CFStringFindWithOptions", referenced from:
      _testTTFOTFSubfamilyName in libui.a(fonttraits.m.o)
  "_CFStringGetLength", referenced from:
      _testTTFOTFSubfamilyName in libui.a(fonttraits.m.o)
  "_CFStringGetRangeOfComposedCharactersAtIndex", referenced from:
      _uiprivNewGraphemes in libui.a(graphemes.m.o)
  "_CGAffineTransformConcat", referenced from:
      _uiDrawMatrixMultiply in libui.a(draw.m.o)
  "_CGAffineTransformEqualToTransform", referenced from:
      _uiDrawMatrixInvertible in libui.a(draw.m.o)
      _uiDrawMatrixInvert in libui.a(draw.m.o)
  "_CGAffineTransformIdentity", referenced from:
      -[uiprivTextFrame draw:textLayout:at:y:] in libui.a(drawtext.m.o)
  "_CGAffineTransformInvert", referenced from:
      _uiDrawMatrixInvertible in libui.a(draw.m.o)
      _uiDrawMatrixInvert in libui.a(draw.m.o)
  "_CGAffineTransformRotate", referenced from:
      _uiDrawMatrixRotate in libui.a(draw.m.o)
  "_CGAffineTransformScale", referenced from:
      _uiDrawMatrixScale in libui.a(draw.m.o)
  "_CGAffineTransformTranslate", referenced from:
      _uiDrawMatrixTranslate in libui.a(draw.m.o)
      _uiDrawMatrixScale in libui.a(draw.m.o)
      _uiDrawMatrixRotate in libui.a(draw.m.o)
  "_CGColorCreate", referenced from:
      _mkcolor in libui.a(attrstr.m.o)
  "_CGColorSpaceCreateWithName", referenced from:
      _mkcolor in libui.a(attrstr.m.o)
      _fillGradient in libui.a(draw.m.o)
  "_CGColorSpaceRelease", referenced from:
      _fillGradient in libui.a(draw.m.o)
  "_CGContextAddPath", referenced from:
      _uiDrawFill in libui.a(draw.m.o)
      _uiDrawClip in libui.a(draw.m.o)
  "_CGContextClip", referenced from:
      _fillGradient in libui.a(draw.m.o)
      _uiDrawClip in libui.a(draw.m.o)
  "_CGContextConcatCTM", referenced from:
      _uiDrawTransform in libui.a(draw.m.o)
  "_CGContextDrawLinearGradient", referenced from:
      _fillGradient in libui.a(draw.m.o)
  "_CGContextDrawRadialGradient", referenced from:
      _fillGradient in libui.a(draw.m.o)
  "_CGContextEOClip", referenced from:
      _fillGradient in libui.a(draw.m.o)
      _uiDrawClip in libui.a(draw.m.o)
  "_CGContextEOFillPath", referenced from:
      _fillSolid in libui.a(draw.m.o)
  "_CGContextFillPath", referenced from:
      _fillSolid in libui.a(draw.m.o)
  "_CGContextGetTextMatrix", referenced from:
      -[uiprivTextFrame draw:textLayout:at:y:] in libui.a(drawtext.m.o)
  "_CGContextRestoreGState", referenced from:
      -[uiprivTextFrame draw:textLayout:at:y:] in libui.a(drawtext.m.o)
      _fillGradient in libui.a(draw.m.o)
      _uiDrawRestore in libui.a(draw.m.o)
  "_CGContextSaveGState", referenced from:
      -[uiprivTextFrame draw:textLayout:at:y:] in libui.a(drawtext.m.o)
      _fillGradient in libui.a(draw.m.o)
      _uiDrawSave in libui.a(draw.m.o)
  "_CGContextScaleCTM", referenced from:
      -[uiprivTextFrame draw:textLayout:at:y:] in libui.a(drawtext.m.o)
  "_CGContextSetRGBFillColor", referenced from:
      _fillSolid in libui.a(draw.m.o)
  "_CGContextSetTextMatrix", referenced from:
      -[uiprivTextFrame draw:textLayout:at:y:] in libui.a(drawtext.m.o)
  "_CGContextTranslateCTM", referenced from:
      -[uiprivTextFrame draw:textLayout:at:y:] in libui.a(drawtext.m.o)
  "_CGGradientCreateWithColorComponents", referenced from:
      _fillGradient in libui.a(draw.m.o)
  "_CGGradientRelease", referenced from:
      _fillGradient in libui.a(draw.m.o)
  "_CGPathAddArc", referenced from:
      _uiDrawPathArcTo in libui.a(draw.m.o)
  "_CGPathAddCurveToPoint", referenced from:
      _uiDrawPathBezierTo in libui.a(draw.m.o)
  "_CGPathAddLineToPoint", referenced from:
      _uiDrawPathLineTo in libui.a(draw.m.o)
  "_CGPathAddRect", referenced from:
      _uiDrawPathAddRectangle in libui.a(draw.m.o)
  "_CGPathCloseSubpath", referenced from:
      _uiDrawPathCloseFigure in libui.a(draw.m.o)
  "_CGPathCreateCopyByDashingPath", referenced from:
      _uiDrawStroke in libui.a(draw.m.o)
  "_CGPathCreateCopyByStrokingPath", referenced from:
      _uiDrawStroke in libui.a(draw.m.o)
  "_CGPathCreateMutable", referenced from:
      _uiDrawNewPath in libui.a(draw.m.o)
  "_CGPathCreateWithRect", referenced from:
      -[uiprivTextFrame initWithLayoutParams:] in libui.a(drawtext.m.o)
  "_CGPathMoveToPoint", referenced from:
      _uiDrawPathNewFigure in libui.a(draw.m.o)
      _uiDrawPathNewFigureWithArc in libui.a(draw.m.o)
  "_CGPathRelease", referenced from:
      _uiDrawFreePath in libui.a(draw.m.o)
      _uiDrawStroke in libui.a(draw.m.o)
  "_CGPointZero", referenced from:
      -[uiprivTextFrame initWithLayoutParams:] in libui.a(drawtext.m.o)
  "_CTFontCopyFontDescriptor", referenced from:
      -[uiprivFontButton getfontdesc:] in libui.a(fontbutton.m.o)
      -[uiprivFontStyleData initWithFont:] in libui.a(fontmatch.m.o)
  "_CTFontCopyName", referenced from:
      -[uiprivFontStyleData fontName:] in libui.a(fontmatch.m.o)
  "_CTFontCopyPostScriptName", referenced from:
      -[uiprivFontStyleData postScriptName] in libui.a(fontmatch.m.o)
  "_CTFontCopyTable", referenced from:
      -[uiprivFontStyleData table:] in libui.a(fontmatch.m.o)
  "_CTFontCopyVariationAxes", referenced from:
      -[uiprivFontStyleData variationAxes] in libui.a(fontmatch.m.o)
  "_CTFontCreateWithFontDescriptor", referenced from:
      -[uiprivCombinedFontAttr toCTFontWithDefaultFont:] in libui.a(attrstr.m.o)
      -[uiprivFontStyleData ensureFont] in libui.a(fontmatch.m.o)
  "_CTFontDescriptorCopyAttribute", referenced from:
      -[uiprivFontStyleData prepare] in libui.a(fontmatch.m.o)
      -[uiprivFontStyleData styleName] in libui.a(fontmatch.m.o)
      -[uiprivFontStyleData variation] in libui.a(fontmatch.m.o)
      _uiprivFontDescriptorFromCTFontDescriptor in libui.a(fontmatch.m.o)
  "_CTFontDescriptorCreateCopyWithAttributes", referenced from:
      _uiprivCTFontDescriptorAppendFeatures in libui.a(fontmatch.m.o)
  "_CTFontDescriptorCreateMatchingFontDescriptors", referenced from:
      _matchStyle in libui.a(fontmatch.m.o)
  "_CTFontDescriptorCreateWithAttributes", referenced from:
      _uiprivFontDescriptorToCTFontDescriptor in libui.a(fontmatch.m.o)
  "_CTFontGetSize", referenced from:
      -[uiprivFontButton getfontdesc:] in libui.a(fontbutton.m.o)
  "_CTFrameDraw", referenced from:
      -[uiprivTextFrame draw:textLayout:at:y:] in libui.a(drawtext.m.o)
  "_CTFrameGetLines", referenced from:
      -[uiprivTextFrame lines] in libui.a(drawtext.m.o)
  "_CTFramesetterCreateFrame", referenced from:
      -[uiprivTextFrame initWithLayoutParams:] in libui.a(drawtext.m.o)
  "_CTFramesetterCreateWithAttributedString", referenced from:
      -[uiprivTextFrame initWithLayoutParams:] in libui.a(drawtext.m.o)
  "_CTFramesetterSuggestFrameSizeWithConstraints", referenced from:
      -[uiprivTextFrame initWithLayoutParams:] in libui.a(drawtext.m.o)
  "_CTParagraphStyleCreate", referenced from:
      _mkParagraphStyle in libui.a(attrstr.m.o)
  "_NSApp", referenced from:
      -[uiprivApplicationClass terminate:] in libui.a(main.m.o)
      _uiInit in libui.a(main.m.o)
      _uiUninit in libui.a(main.m.o)
      _uiMain in libui.a(main.m.o)
      ___uiMain_block_invoke in libui.a(main.m.o)
      _uiMainSteps in libui.a(main.m.o)
      _uiprivMainStep in libui.a(main.m.o)
      ...
     (maybe you meant: __OBJC_PROTOCOL_$_NSApplicationDelegate, __OBJC_LABEL_PROTOCOL_$_NSApplicationDelegate )
  "_NSDefaultRunLoopMode", referenced from:
      _uiMainStep in libui.a(main.m.o)
  "_NSDisableScreenUpdates", referenced from:
      _minMaxAutoLayoutSizes in libui.a(winmoveresize.m.o)
  "_NSEnableScreenUpdates", referenced from:
      _minMaxAutoLayoutSizes in libui.a(winmoveresize.m.o)
  "_NSEndMapTableEnumeration", referenced from:
      _uiprivMapWalk in libui.a(map.m.o)
  "_NSEnumerateMapTable", referenced from:
      _uiprivMapWalk in libui.a(map.m.o)
  "_NSEventTrackingRunLoopMode", referenced from:
      _uiprivDoManualMove in libui.a(winmoveresize.m.o)
      _uiprivDoManualResize in libui.a(winmoveresize.m.o)
  "_NSLog", referenced from:
      -[uiprivAppDelegate applicationShouldTerminate:] in libui.a(main.m.o)
      _uiprivLoadUndocumented in libui.a(undocumented.m.o)
      _uiprivRealBug in libui.a(debug.m.o)
  "_NSMapGet", referenced from:
      _uiprivMapGet in libui.a(map.m.o)
  "_NSMapInsert", referenced from:
      _uiprivMapSet in libui.a(map.m.o)
  "_NSMapRemove", referenced from:
      _uiprivMapDelete in libui.a(map.m.o)
  "_NSNextMapEnumeratorPair", referenced from:
      _uiprivMapWalk in libui.a(map.m.o)
  "_NSResetMapTable", referenced from:
      _uiprivMapReset in libui.a(map.m.o)
  "_NSWindowWillCloseNotification", referenced from:
      -[uiprivFontButton activateFontButton] in libui.a(fontbutton.m.o)
      -[uiprivFontButton deactivateFontButton:] in libui.a(fontbutton.m.o)
      -[colorButton activate:] in libui.a(colorbutton.m.o)
      -[colorButton deactivate] in libui.a(colorbutton.m.o)
  "_NSZeroPoint", referenced from:
      -[uiprivApplicationClass terminate:] in libui.a(main.m.o)
  "_NSZeroRect", referenced from:
      _uiNewArea in libui.a(area.m.o)
      -[boxView initWithVertical:b:] in libui.a(box.m.o)
      _uiNewCombobox in libui.a(combobox.m.o)
      _uiNewFontButton in libui.a(fontbutton.m.o)
      -[formChild initWithLabel:] in libui.a(form.m.o)
      -[formView initWithF:] in libui.a(form.m.o)
      _uiNewColorButton in libui.a(colorbutton.m.o)
      ...
  "_NSZeroSize", referenced from:
      _makeIndependent in libui.a(winmoveresize.m.o)
  "_OBJC_CLASS_$_NSApplication", referenced from:
      _OBJC_CLASS_$_uiprivApplicationClass in libui.a(main.m.o)
  "_OBJC_CLASS_$_NSArrayController", referenced from:
      objc-class-ref in libui.a(combobox.m.o)
  "_OBJC_CLASS_$_NSAutoreleasePool", referenced from:
      objc-class-ref in libui.a(main.m.o)
  "_OBJC_CLASS_$_NSButton", referenced from:
      _OBJC_CLASS_$_uiprivFontButton in libui.a(fontbutton.m.o)
  "_OBJC_CLASS_$_NSColor", referenced from:
      objc-class-ref in libui.a(area.m.o)
      objc-class-ref in libui.a(attrstr.m.o)
      objc-class-ref in libui.a(colorbutton.m.o)
  "_OBJC_CLASS_$_NSColorPanel", referenced from:
      objc-class-ref in libui.a(colorbutton.m.o)
  "_OBJC_CLASS_$_NSColorSpace", referenced from:
      objc-class-ref in libui.a(colorbutton.m.o)
  "_OBJC_CLASS_$_NSColorWell", referenced from:
      _OBJC_CLASS_$_colorButton in libui.a(colorbutton.m.o)
  "_OBJC_CLASS_$_NSDate", referenced from:
      objc-class-ref in libui.a(main.m.o)
      objc-class-ref in libui.a(winmoveresize.m.o)
  "_OBJC_CLASS_$_NSEvent", referenced from:
      objc-class-ref in libui.a(area.m.o)
      objc-class-ref in libui.a(main.m.o)
  "_OBJC_CLASS_$_NSFont", referenced from:
      objc-class-ref in libui.a(fontbutton.m.o)
      objc-class-ref in libui.a(control.m.o)
  "_OBJC_CLASS_$_NSFontManager", referenced from:
      objc-class-ref in libui.a(fontbutton.m.o)
  "_OBJC_CLASS_$_NSFontPanel", referenced from:
      _OBJC_CLASS_$_uiprivNonModalFontPanel in libui.a(fontbutton.m.o)
      objc-class-ref in libui.a(fontbutton.m.o)
  "_OBJC_CLASS_$_NSGraphicsContext", referenced from:
      objc-class-ref in libui.a(area.m.o)
      objc-class-ref in libui.a(main.m.o)
  "_OBJC_CLASS_$_NSImage", referenced from:
      objc-class-ref in libui.a(attrstr.m.o)
  "_OBJC_CLASS_$_NSLayoutConstraint", referenced from:
      objc-class-ref in libui.a(autolayout.m.o)
  "_OBJC_CLASS_$_NSMapTable", referenced from:
      objc-class-ref in libui.a(map.m.o)
  "_OBJC_CLASS_$_NSMenu", referenced from:
      objc-class-ref in libui.a(menu.m.o)
  "_OBJC_CLASS_$_NSMenuItem", referenced from:
      objc-class-ref in libui.a(menu.m.o)
  "_OBJC_CLASS_$_NSMutableArray", referenced from:
      objc-class-ref in libui.a(box.m.o)
      objc-class-ref in libui.a(form.m.o)
      objc-class-ref in libui.a(menu.m.o)
      objc-class-ref in libui.a(alloc.m.o)
      objc-class-ref in libui.a(attrstr.m.o)
  "_OBJC_CLASS_$_NSMutableDictionary", referenced from:
      objc-class-ref in libui.a(fontvariation.m.o)
  "_OBJC_CLASS_$_NSMutableString", referenced from:
      objc-class-ref in libui.a(alloc.m.o)
      objc-class-ref in libui.a(debug.m.o)
  "_OBJC_CLASS_$_NSNotificationCenter", referenced from:
      objc-class-ref in libui.a(fontbutton.m.o)
      objc-class-ref in libui.a(colorbutton.m.o)
  "_OBJC_CLASS_$_NSNumber", referenced from:
      objc-class-ref in libui.a(attrstr.m.o)
      objc-class-ref in libui.a(fontvariation.m.o)
  "_OBJC_CLASS_$_NSObject", referenced from:
      _OBJC_CLASS_$_boxChild in libui.a(box.m.o)
      _OBJC_CLASS_$_comboboxDelegateClass in libui.a(combobox.m.o)
      _OBJC_CLASS_$_uiprivDrawTextBackgroundParams in libui.a(drawtext.m.o)
      _OBJC_CLASS_$_uiprivTextFrame in libui.a(drawtext.m.o)
      _OBJC_CLASS_$_uiprivAppDelegate in libui.a(main.m.o)
      _OBJC_CLASS_$_uiprivTimerDelegate in libui.a(main.m.o)
      _OBJC_CLASS_$_windowDelegateClass in libui.a(window.m.o)
      ...
  "_OBJC_CLASS_$_NSPopUpButton", referenced from:
      objc-class-ref in libui.a(combobox.m.o)
  "_OBJC_CLASS_$_NSProcessInfo", referenced from:
      objc-class-ref in libui.a(main.m.o)
      objc-class-ref in libui.a(menu.m.o)
  "_OBJC_CLASS_$_NSScrollView", referenced from:
      objc-class-ref in libui.a(form.m.o)
      objc-class-ref in libui.a(scrollview.m.o)
  "_OBJC_CLASS_$_NSSearchField", referenced from:
      _OBJC_CLASS_$_libui_intrinsicWidthNSSearchField in libui.a(entry.m.o)
      objc-class-ref in libui.a(entry.m.o)
  "_OBJC_CLASS_$_NSSecureTextField", referenced from:
      _OBJC_CLASS_$_libui_intrinsicWidthNSSecureTextField in libui.a(entry.m.o)
  "_OBJC_CLASS_$_NSString", referenced from:
      objc-class-ref in libui.a(combobox.m.o)
      objc-class-ref in libui.a(fontbutton.m.o)
      objc-class-ref in libui.a(form.m.o)
      objc-class-ref in libui.a(window.m.o)
      objc-class-ref in libui.a(menu.m.o)
      objc-class-ref in libui.a(alloc.m.o)
      objc-class-ref in libui.a(attrstr.m.o)
      ...
  "_OBJC_CLASS_$_NSTextField", referenced from:
      objc-class-ref in libui.a(label.m.o)
      _OBJC_CLASS_$_libui_intrinsicWidthNSTextField in libui.a(entry.m.o)
  "_OBJC_CLASS_$_NSTimer", referenced from:
      objc-class-ref in libui.a(main.m.o)
  "_OBJC_CLASS_$_NSTrackingArea", referenced from:
      objc-class-ref in libui.a(area.m.o)
  "_OBJC_CLASS_$_NSValue", referenced from:
      objc-class-ref in libui.a(menu.m.o)
      objc-class-ref in libui.a(alloc.m.o)
  "_OBJC_CLASS_$_NSView", referenced from:
      _OBJC_CLASS_$_areaView in libui.a(area.m.o)
      _OBJC_CLASS_$_boxView in libui.a(box.m.o)
      _OBJC_CLASS_$_formChild in libui.a(form.m.o)
      _OBJC_CLASS_$_formView in libui.a(form.m.o)
  "_OBJC_CLASS_$_NSWindow", referenced from:
      _OBJC_CLASS_$_uiprivNSWindow in libui.a(window.m.o)
  "_OBJC_METACLASS_$_NSApplication", referenced from:
      _OBJC_METACLASS_$_uiprivApplicationClass in libui.a(main.m.o)
  "_OBJC_METACLASS_$_NSButton", referenced from:
      _OBJC_METACLASS_$_uiprivFontButton in libui.a(fontbutton.m.o)
  "_OBJC_METACLASS_$_NSColorWell", referenced from:
      _OBJC_METACLASS_$_colorButton in libui.a(colorbutton.m.o)
  "_OBJC_METACLASS_$_NSFontPanel", referenced from:
      _OBJC_METACLASS_$_uiprivNonModalFontPanel in libui.a(fontbutton.m.o)
  "_OBJC_METACLASS_$_NSObject", referenced from:
      _OBJC_METACLASS_$_areaView in libui.a(area.m.o)
      _OBJC_METACLASS_$_boxChild in libui.a(box.m.o)
      _OBJC_METACLASS_$_boxView in libui.a(box.m.o)
      _OBJC_METACLASS_$_comboboxDelegateClass in libui.a(combobox.m.o)
      _OBJC_METACLASS_$_uiprivDrawTextBackgroundParams in libui.a(drawtext.m.o)
      _OBJC_METACLASS_$_uiprivTextFrame in libui.a(drawtext.m.o)
      _OBJC_METACLASS_$_uiprivFontButton in libui.a(fontbutton.m.o)
      ...
  "_OBJC_METACLASS_$_NSSearchField", referenced from:
      _OBJC_METACLASS_$_libui_intrinsicWidthNSSearchField in libui.a(entry.m.o)
  "_OBJC_METACLASS_$_NSSecureTextField", referenced from:
      _OBJC_METACLASS_$_libui_intrinsicWidthNSSecureTextField in libui.a(entry.m.o)
  "_OBJC_METACLASS_$_NSTextField", referenced from:
      _OBJC_METACLASS_$_libui_intrinsicWidthNSTextField in libui.a(entry.m.o)
  "_OBJC_METACLASS_$_NSView", referenced from:
      _OBJC_METACLASS_$_areaView in libui.a(area.m.o)
      _OBJC_METACLASS_$_boxView in libui.a(box.m.o)
      _OBJC_METACLASS_$_formChild in libui.a(form.m.o)
      _OBJC_METACLASS_$_formView in libui.a(form.m.o)
  "_OBJC_METACLASS_$_NSWindow", referenced from:
      _OBJC_METACLASS_$_uiprivNSWindow in libui.a(window.m.o)
  "___CFConstantStringClassReference", referenced from:
      CFString in libui.a(box.m.o)
      CFString in libui.a(box.m.o)
      CFString in libui.a(box.m.o)
      CFString in libui.a(box.m.o)
      CFString in libui.a(box.m.o)
      CFString in libui.a(box.m.o)
      CFString in libui.a(box.m.o)
      ...
  "___objc_personality_v0", referenced from:
      _uiprivDoManualMove in libui.a(winmoveresize.m.o)
      _uiprivDoManualResize in libui.a(winmoveresize.m.o)
      Dwarf Exception Unwind Info (__eh_frame) in libui.a(winmoveresize.m.o)
      _otfArrayForEachAAT in libui.a(opentype.m.o)
      Dwarf Exception Unwind Info (__eh_frame) in libui.a(opentype.m.o)
  "__objc_empty_cache", referenced from:
      _OBJC_CLASS_$_areaView in libui.a(area.m.o)
      _OBJC_METACLASS_$_areaView in libui.a(area.m.o)
      _OBJC_METACLASS_$_boxChild in libui.a(box.m.o)
      _OBJC_CLASS_$_boxChild in libui.a(box.m.o)
      _OBJC_CLASS_$_boxView in libui.a(box.m.o)
      _OBJC_METACLASS_$_boxView in libui.a(box.m.o)
      _OBJC_CLASS_$_comboboxDelegateClass in libui.a(combobox.m.o)
      ...
  "_kCFAllocatorNull", referenced from:
      _uiprivNewGraphemes in libui.a(graphemes.m.o)
  "_kCFCopyStringDictionaryKeyCallBacks", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
      _uiprivFontDescriptorToCTFontDescriptor in libui.a(fontmatch.m.o)
      _uiprivCTFontDescriptorAppendFeatures in libui.a(fontmatch.m.o)
      _addCTFeatureEntry in libui.a(opentype.m.o)
  "_kCFTypeArrayCallBacks", referenced from:
      _uiprivOpenTypeFeaturesToCTFeatures in libui.a(opentype.m.o)
  "_kCFTypeDictionaryValueCallBacks", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
      _uiprivFontDescriptorToCTFontDescriptor in libui.a(fontmatch.m.o)
      _uiprivCTFontDescriptorAppendFeatures in libui.a(fontmatch.m.o)
      _addCTFeatureEntry in libui.a(opentype.m.o)
  "_kCGColorSpaceSRGB", referenced from:
      _mkcolor in libui.a(attrstr.m.o)
      _fillGradient in libui.a(draw.m.o)
  "_kCTFontAttributeName", referenced from:
      _applyFontAttributes in libui.a(attrstr.m.o)
  "_kCTFontFamilyNameAttribute", referenced from:
      _uiprivFontDescriptorToCTFontDescriptor in libui.a(fontmatch.m.o)
      _uiprivFontDescriptorFromCTFontDescriptor in libui.a(fontmatch.m.o)
  "_kCTFontFamilyNameKey", referenced from:
      -[uiprivFontStyleData familyName] in libui.a(fontmatch.m.o)
  "_kCTFontFeatureSelectorIdentifierKey", referenced from:
      ___otfArrayForEachAAT_block_invoke in libui.a(opentype.m.o)
  "_kCTFontFeatureSettingsAttribute", referenced from:
      _uiprivCTFontDescriptorAppendFeatures in libui.a(fontmatch.m.o)
  "_kCTFontFeatureTypeIdentifierKey", referenced from:
      ___otfArrayForEachAAT_block_invoke in libui.a(opentype.m.o)
  "_kCTFontFormatAttribute", referenced from:
      -[uiprivFontStyleData prepare] in libui.a(fontmatch.m.o)
  "_kCTFontFullNameKey", referenced from:
      -[uiprivFontStyleData fullName] in libui.a(fontmatch.m.o)
  "_kCTFontRegistrationScopeAttribute", referenced from:
      -[uiprivFontStyleData prepare] in libui.a(fontmatch.m.o)
  "_kCTFontSizeAttribute", referenced from:
      _uiprivFontDescriptorToCTFontDescriptor in libui.a(fontmatch.m.o)
  "_kCTFontStyleNameAttribute", referenced from:
      -[uiprivFontStyleData styleName] in libui.a(fontmatch.m.o)
  "_kCTFontSubFamilyNameKey", referenced from:
      -[uiprivFontStyleData subFamilyName] in libui.a(fontmatch.m.o)
  "_kCTFontSymbolicTrait", referenced from:
      -[uiprivFontStyleData prepare] in libui.a(fontmatch.m.o)
  "_kCTFontTraitsAttribute", referenced from:
      -[uiprivFontStyleData prepare] in libui.a(fontmatch.m.o)
  "_kCTFontVariationAttribute", referenced from:
      -[uiprivFontStyleData variation] in libui.a(fontmatch.m.o)
  "_kCTFontVariationAxisDefaultValueKey", referenced from:
      -[fvarAxis initWithIndex:dict:avarTable:] in libui.a(fontvariation.m.o)
  "_kCTFontVariationAxisIdentifierKey", referenced from:
      _uiprivMakeVariationAxisDict in libui.a(fontvariation.m.o)
  "_kCTFontVariationAxisMaximumValueKey", referenced from:
      -[fvarAxis initWithIndex:dict:avarTable:] in libui.a(fontvariation.m.o)
  "_kCTFontVariationAxisMinimumValueKey", referenced from:
      -[fvarAxis initWithIndex:dict:avarTable:] in libui.a(fontvariation.m.o)
  "_kCTFontWeightTrait", referenced from:
      -[uiprivFontStyleData prepare] in libui.a(fontmatch.m.o)
  "_kCTFontWidthTrait", referenced from:
      -[uiprivFontStyleData prepare] in libui.a(fontmatch.m.o)
  "_kCTForegroundColorAttributeName", referenced from:
      _processAttribute in libui.a(attrstr.m.o)
  "_kCTParagraphStyleAttributeName", referenced from:
      _uiprivAttributedStringToCFAttributedString in libui.a(attrstr.m.o)
  "_kCTUnderlineColorAttributeName", referenced from:
      _processAttribute in libui.a(attrstr.m.o)
  "_kCTUnderlineStyleAttributeName", referenced from:
      _processAttribute in libui.a(attrstr.m.o)
  "_objc_autoreleasePoolPop", referenced from:
      _uiInit in libui.a(main.m.o)
      _uiUninit in libui.a(main.m.o)
      _uiprivMainStep in libui.a(main.m.o)
      _newItem in libui.a(menu.m.o)
      _uiNewMenu in libui.a(menu.m.o)
      -[uiprivCombinedFontAttr hash] in libui.a(attrstr.m.o)
  "_objc_autoreleasePoolPush", referenced from:
      _uiInit in libui.a(main.m.o)
      _uiUninit in libui.a(main.m.o)
      _uiprivMainStep in libui.a(main.m.o)
      _newItem in libui.a(menu.m.o)
      _uiNewMenu in libui.a(menu.m.o)
      -[uiprivCombinedFontAttr hash] in libui.a(attrstr.m.o)
  "_objc_enumerationMutation", referenced from:
      -[boxView onDestroy] in libui.a(box.m.o)
      -[boxView syncEnableStates:] in libui.a(box.m.o)
      -[boxView establishOurConstraints] in libui.a(box.m.o)
      -[boxView setPadded:] in libui.a(box.m.o)
      -[boxView nStretchy] in libui.a(box.m.o)
      -[uiprivTextFrame draw:textLayout:at:y:] in libui.a(drawtext.m.o)
      -[formView onDestroy] in libui.a(form.m.o)
      ...
  "_objc_getProperty", referenced from:
      -[formChild label] in libui.a(form.m.o)
      -[formChild baseline] in libui.a(form.m.o)
      -[formChild leading] in libui.a(form.m.o)
      -[formChild top] in libui.a(form.m.o)
      -[formChild trailing] in libui.a(form.m.o)
      -[formChild bottom] in libui.a(form.m.o)
      -[uiprivAppDelegate menuManager] in libui.a(main.m.o)
      ...
  "_objc_msgSend", referenced from:
      -[areaView initWithFrame:area:] in libui.a(area.m.o)
      -[areaView drawRect:] in libui.a(area.m.o)
      -[areaView parseModifiers:] in libui.a(area.m.o)
      -[areaView setupNewTrackingArea] in libui.a(area.m.o)
      -[areaView updateTrackingAreas] in libui.a(area.m.o)
      -[areaView doMouseEvent:] in libui.a(area.m.o)
      -[areaView mouseMoved:] in libui.a(area.m.o)
      ...
  "_objc_msgSendSuper2", referenced from:
      -[areaView initWithFrame:area:] in libui.a(area.m.o)
      -[areaView setFrameSize:] in libui.a(area.m.o)
      -[areaView intrinsicContentSize] in libui.a(area.m.o)
      -[boxView initWithVertical:b:] in libui.a(box.m.o)
      -[comboboxDelegateClass init] in libui.a(combobox.m.o)
      -[comboboxDelegateClass dealloc] in libui.a(combobox.m.o)
      -[uiprivDrawTextBackgroundParams initWithStart:end:r:g:b:a:] in libui.a(drawtext.m.o)
      ...
  "_objc_msgSend_stret", referenced from:
      -[areaView drawRect:] in libui.a(area.m.o)
      -[areaView setupNewTrackingArea] in libui.a(area.m.o)
      -[areaView doMouseEvent:] in libui.a(area.m.o)
      _uiWindowContentSize in libui.a(window.m.o)
      _makeIndependent in libui.a(winmoveresize.m.o)
      _uiprivDoManualMove in libui.a(winmoveresize.m.o)
      _uiprivDoManualResize in libui.a(winmoveresize.m.o)
      ...
  "_objc_setProperty_atomic", referenced from:
      -[formChild setLabel:] in libui.a(form.m.o)
      -[formChild setBaseline:] in libui.a(form.m.o)
      -[formChild setLeading:] in libui.a(form.m.o)
      -[formChild setTop:] in libui.a(form.m.o)
      -[formChild setTrailing:] in libui.a(form.m.o)
      -[formChild setBottom:] in libui.a(form.m.o)
      -[uiprivAppDelegate setMenuManager:] in libui.a(main.m.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [pp] Error 1
make[1]: *** [CMakeFiles/pp.dir/all] Error 2
make: *** [all] Error 2
```

**Proposed solution**
Would be nice if the example would build

**Describe alternatives you've considered**
I'm trying other cross-platform GUI frameworks that vcpkg 'supports', e.g., see the trackbacks at https://github.com/microsoft/vcpkg/issues/17052

--- a/ports/libui/003-fix-system-link.patch
+++ b/ports/libui/003-fix-system-link.patch
@@ -1,0 +1,17 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index a23b84d..9892dfc 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -172,6 +172,12 @@ if(BUILD_SHARED_LIBS)
+ 	target_link_libraries(libui
+ 		PRIVATE ${_LIBUI_LIBS})
+ endif()
++
++if (APPLE)
++    find_library(Foundation Foundation)
++    find_library(AppKit AppKit)
++    target_link_libraries(libui PUBLIC $<$<PLATFORM_ID:Darwin>:${Foundation};${AppKit}>)
++endif()
+ # TODO INTERFACE libs don't inherit to grandhcildren?
+ # on Windows the linker for static libraries is different; don't give it the flags
+ if(BUILD_SHARED_LIBS)

--- a/ports/libui/CONTROL
+++ b/ports/libui/CONTROL
@@ -1,3 +1,0 @@
-Source: libui
-Version: 2018-11-03-1
-Description: Simple and portable (but not inflexible) native GUI library in C.

--- a/ports/libui/portfile.cmake
+++ b/ports/libui/portfile.cmake
@@ -24,4 +24,4 @@ vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/libui TARGET_PATH share/unoffici
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 # Handle copyright
-configure_file("${SOURCE_PATH}/LICENSE" "${CURRENT_PACKAGES_DIR}/share/libui/copyright" COPYONLY)
+configure_file("${SOURCE_PATH}/LICENSE" "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" COPYONLY)

--- a/ports/libui/portfile.cmake
+++ b/ports/libui/portfile.cmake
@@ -7,6 +7,7 @@ vcpkg_from_github(
     PATCHES
         "001-fix-cmake.patch"
         "002-fix-macosx-build.patch"
+        "003-fix-system-link.patch"
 )
 
 vcpkg_configure_cmake(
@@ -16,11 +17,11 @@ vcpkg_configure_cmake(
 
 vcpkg_install_cmake()
 
+vcpkg_copy_pdbs()
+
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/libui TARGET_PATH share/unofficial-libui)
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 # Handle copyright
-configure_file(${SOURCE_PATH}/LICENSE ${CURRENT_PACKAGES_DIR}/share/libui/copyright COPYONLY)
-
-vcpkg_copy_pdbs()
+configure_file("${SOURCE_PATH}/LICENSE" "${CURRENT_PACKAGES_DIR}/share/libui/copyright" COPYONLY)

--- a/ports/libui/vcpkg.json
+++ b/ports/libui/vcpkg.json
@@ -1,0 +1,7 @@
+{
+  "name": "libui",
+  "version-date": "2018-11-03",
+  "port-version": 2,
+  "description": "Simple and portable (but not inflexible) native GUI library in C.",
+  "homepage": "https://github.com/andlabs/libui"
+}

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3537,8 +3537,8 @@
       "port-version": 0
     },
     "libui": {
-      "baseline": "2018-11-03-1",
-      "port-version": 0
+      "baseline": "2018-11-03",
+      "port-version": 2
     },
     "libunibreak": {
       "baseline": "4.3-0",

--- a/versions/l-/libui.json
+++ b/versions/l-/libui.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "302766b82a15672a75618bad9bb8e8be58ef6ba8",
+      "git-tree": "941bfc035a8cfa6020be3475769732b027249a1d",
       "version-date": "2018-11-03",
       "port-version": 2
     },

--- a/versions/l-/libui.json
+++ b/versions/l-/libui.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "302766b82a15672a75618bad9bb8e8be58ef6ba8",
+      "version-date": "2018-11-03",
+      "port-version": 2
+    },
+    {
       "git-tree": "c3d3ba8694ca2465c505735531cd10dfc8cf150d",
       "version-string": "2018-11-03-1",
       "port-version": 0

